test/libgit2.jl

```diff
@@ -567,6 +570,178 @@ mktempdir() do dir
         @test creds.user == creds_user
         @test creds.pass == creds_pass
     #end
+
+    #@testset "SSH" begin
+        sshd_command = ""
+        ssh_repo = joinpath(dir, "Example.SSH")
+        if !is_windows()
+            try
+                sshd_command = strip(readstring(`which sshd`))
+            catch
+                warn("Skipping SSH tests because (Are `which` and `sshd` installed?)")
+            end
+        end
+        if !isempty(sshd_command)
+            mktempdir() do fakehomedir
+                mkdir(joinpath(fakehomedir,".ssh"))
+                # Unsetting the SSH agent serves two purposes. First, we make
+                # sure that we don't accidentally pick up an existing agent,
+                # and second we test that we fall back to using a key file
+                # if the agent isn't present.
+                withenv("HOME"=>fakehomedir,"SSH_AUTH_SOCK"=>nothing) do
+                    # Generate user file, first an unencrypted one
+                    wait(spawn(`ssh-keygen -N "" -C juliatest@localhost -f $fakehomedir/.ssh/id_rsa`))
+
+                    # Generate host keys
+                    wait(spawn(`ssh-keygen -f $fakehomedir/ssh_host_rsa_key -N '' -t rsa`))
+                    wait(spawn(`ssh-keygen -f $fakehomedir/ssh_host_dsa_key -N '' -t dsa`))
+
+                    our_ssh_port = rand(13000:14000) # Chosen arbitrarily
+
+                    key_option = "AuthorizedKeysFile $fakehomedir/.ssh/id_rsa.pub"
+                    pidfile_option = "PidFile $fakehomedir/sshd.pid"
+                    sshp = agentp = nothing
+                    logfile = tempname()
+                    ssh_debug = false
+                    function spawn_sshd()
+                        debug_flags = ssh_debug ? `-d -d` : ``
+                        _p = spawn(pipeline(pipeline(`$sshd_command -f /dev/null
+                        $debug_flags
+                        -h $fakehomedir/ssh_host_rsa_key
+                        -h $fakehomedir/ssh_host_dsa_key -p $our_ssh_port
+                        -E $logfile
+                        -o $pidfile_option
+                        -o 'Protocol 2'
+                        -o $key_option
+                        -o 'UsePrivilegeSeparation no'
+                        -o 'StrictModes no'`,STDOUT),stderr=STDERR))
+                        # Give the SSH server 5 seconds to start up
+                        yield(); sleep(5)
+                        _p
+                    end
+                    sshp = spawn_sshd()
+
+                    TIOCSCTTY_str = "ccall(:ioctl, Void, (Cint, Cint, Int64), 0,
+                        is_apple() ? 0x20007461 : is_linux() ? 0x540E :
+                        error(\"Fill in TIOCSCTTY for this OS here\"), 0)"
```

Comment:
yeah, unless you a) tell me the value of TIOCSCTTY on freebsd and other relevant platforms or b) prevent @vtjnash from merging the patch that makes this necessary.